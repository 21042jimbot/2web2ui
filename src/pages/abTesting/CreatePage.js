import React, { Component } from 'react';
import { connect } from 'react-redux';
import { LINKS } from 'src/constants';
import { setSubaccountQuery } from 'src/helpers/subaccounts';
import { HibanaConsumer } from 'src/context/HibanaContext';

// Actions
import { createAbTestDraft } from 'src/actions/abTesting';
import { showAlert } from 'src/actions/globalAlert';
import { listTemplates } from 'src/actions/templates';

// Components
import { PageLink } from 'src/components/links';
import { Panel, Page, Text } from 'src/components/matchbox';
import AbTestCreateForm from './components/AbTestCreateForm';
export class CreatePage extends Component {
  componentDidMount() {
    // Get templates here for the typeahead
    // Ensures the list is always up to date
    this.props.listTemplates();
  }

  create = values => {
    const { createAbTestDraft, showAlert, history } = this.props;
    const { id, name, subaccount, default_variant = {} } = values;
    const subaccountId = subaccount ? subaccount.id : null;
    const default_template = { template_id: default_variant.id };
    const abTest = { id, name, default_template };

    return createAbTestDraft({ abTest, subaccount: subaccountId }).then(() => {
      showAlert({ type: 'success', message: 'A/B test draft created' });
      history.push(`/ab-testing/${abTest.id}/1${setSubaccountQuery(subaccountId)}`);
    });
  };

  render_hibana = () => (
    <Page
      breadcrumbAction={{ content: 'Back to A/B Tests', component: PageLink, to: '/ab-testing' }}
    >
      <Text as="h1" fontWeight="500" marginTop={-15} mb={20}>
        Create a New A/B Test
      </Text>
      <Panel>
        <AbTestCreateForm onSubmit={this.create} />
      </Panel>
    </Page>
  );

  render() {
    return (
      <HibanaConsumer>
        {({ isHibanaEnabled }) => (
          <>
            {isHibanaEnabled ? (
              this.render_hibana()
            ) : (
              <Page
                breadcrumbAction={{
                  content: 'Back to A/B Tests',
                  component: PageLink,
                  to: '/ab-testing',
                }}
              >
                <Panel
                  title="Create a New A/B Test"
                  actions={[
                    {
                      content: 'Learn more about A/B tests',
                      color: 'orange',
                      to: LINKS.AB_TESTING_API,
                      external: true,
                    },
                  ]}
                >
                  <AbTestCreateForm onSubmit={this.create} />
                </Panel>
              </Page>
            )}
          </>
        )}
      </HibanaConsumer>
    );
  }
}

export default connect(null, { createAbTestDraft, showAlert, listTemplates })(CreatePage);
