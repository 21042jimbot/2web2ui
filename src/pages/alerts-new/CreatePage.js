import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { Page } from '@sparkpost/matchbox';
import withAlertsCreate from './containers/CreatePage.container';
import AlertForm from './components/AlertForm';
import _ from 'lodash';
import formatActionData from './helpers/formatActionData';

export class CreatePage extends Component {

  /*
    Passed as onSubmit to AlertForm. Figures out what updates need to be passed
    to the createAlert action.
  */
  handleCreate = async (values) => {
    const { createAlert, showAlert } = this.props;
    const alertBody = formatActionData(values);

    if ((alertBody.alert_metric === 'signals_health_dod') || ('signals_health_wow')) {
      const dropsValue = alertBody.threshold.error.target;
      const negDropsValue = (dropsValue > 0) ? dropsValue * (-1) : dropsValue;
      alertBody.threshold.error.target = negDropsValue;
    }

    await createAlert({
      data: _.omit(alertBody, 'id', 'subaccount', 'assignTo')
    }).then((response) => {
      showAlert({ type: 'success', message: 'Alert created' });
      this.props.history.push(`/alerts-new/edit/${response.id}`);
    });
  }

  render() {
    return (
      <Page
        title='Create Alert'
        breadcrumbAction={{ content: 'Back to Alerts', to: '/alerts', component: Link }}>
        <AlertForm newAlert = {true} submitting={this.props.loading} onSubmit = {this.handleCreate}/>
      </Page>
    );
  }
}

export default withAlertsCreate(CreatePage);
