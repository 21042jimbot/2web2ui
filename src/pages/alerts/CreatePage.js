import React, { Component } from 'react';
import { Page } from '@sparkpost/matchbox';
import withAlertsCreate from './containers/CreatePage.container';
import AlertForm from './components/AlertForm';
import formatFormValues from './helpers/formatFormValues';
import { Loading } from 'src/components';
import { PageLink } from 'src/components/links';
import { RedirectAndAlert } from 'src/components/globalAlert';

export class CreatePage extends Component {
  componentDidMount() {
    const { getAlert, idToDuplicate } = this.props;
    if (idToDuplicate) {
      getAlert({ id: idToDuplicate });
    }
  }

  handleCreate = values => {
    const { createAlert, showUIAlert, history } = this.props;
    return createAlert({
      data: formatFormValues(values),
    }).then(({ id }) => {
      showUIAlert({ type: 'success', message: 'Alert created' });
      history.push(`/alerts/details/${id}`);
    });
  };

  render() {
    const { loading, getError, getLoading, idToDuplicate } = this.props;

    if (getLoading) {
      return <Loading />;
    }

    if (getError) {
      return <RedirectAndAlert to="/alerts" alert={{ type: 'error', message: getError.message }} />;
    }

    const backBreadcrumb = idToDuplicate
      ? { content: 'Back to Alert', to: `/alerts/details/${idToDuplicate}` }
      : { content: 'Back to Alerts', to: '/alerts' };
    return (
      <Page title="Create Alert" breadcrumbAction={{ ...backBreadcrumb, component: PageLink }}>
        <AlertForm
          submitting={loading}
          onSubmit={this.handleCreate}
          isDuplicate={Boolean(idToDuplicate)}
          isNewAlert={true}
        />
      </Page>
    );
  }
}

export default withAlertsCreate(CreatePage);
