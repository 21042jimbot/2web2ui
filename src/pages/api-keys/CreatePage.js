import React from 'react';
import { connect } from 'react-redux';
import { Link, withRouter } from 'react-router-dom';

import { createApiKey, listGrants, listSubaccountGrants } from 'src/actions/api-keys';
import { showAlert } from 'src/actions/globalAlert';
import { getFormLoading } from 'src/selectors/api-keys';
import { hasSubaccounts } from 'src/selectors/subaccounts';

import ApiKeyForm from './components/ApiKeyForm';
import { Page, Panel } from '@sparkpost/matchbox';
import { Loading } from 'src/components';

const breadcrumbAction = {
  content: 'API Keys',
  Component: Link,
  to: '/account/api-keys'
};

export class CreatePage extends React.Component {
  componentDidMount() {
    this.props.listGrants();
    if (this.props.hasSubaccounts) {
      this.props.listSubaccountGrants();
    }
  }

  onSubmit = (values) => {
    const { createApiKey, history, showAlert } = this.props;

    return createApiKey(values).then((res) => {
      showAlert({ type: 'success', message: 'API key created' });
      history.push('/account/api-keys');
    });
  };

  render() {

    if (this.props.loading) {
      return <Loading />;
    }

    return (
      <Page title="Create API Key" breadcrumbAction={breadcrumbAction}>
        <Panel>
          <ApiKeyForm onSubmit={this.onSubmit} />
        </Panel>
      </Page>
    );
  }
}

const mapStateToProps = (state, props) => ({
  loading: getFormLoading(state),
  hasSubaccounts: hasSubaccounts(state)
});

export default withRouter(
  connect(mapStateToProps, { createApiKey, listGrants, listSubaccountGrants, showAlert })(CreatePage)
);
