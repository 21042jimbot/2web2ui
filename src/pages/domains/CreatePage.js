import React from 'react';
import { Page } from 'src/components/matchbox';
import Domains from './components';

export default function CreatePage() {
  return (
    <Domains.Container>
      <Page title="Add a Domain"></Page>
    </Domains.Container>
  );
}
