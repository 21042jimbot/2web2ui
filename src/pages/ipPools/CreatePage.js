import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import { Page } from '@sparkpost/matchbox';
import { ApiErrorBanner, Loading } from 'src/components';
import PoolForm from './components/PoolForm';

import { showAlert } from 'src/actions/globalAlert';
import { createPool, listPools } from 'src/actions/ipPools';


const breadcrumbAction = {
  content: 'IP Pools',
  Component: Link,
  to: '/account/ip-pools'
};

export class CreatePage extends Component {
  loadDependentData = () => {
    this.props.listPools();
  };

  componentDidMount() {
    this.loadDependentData();
  }

  createPool = (values) => {
    const { createPool, showAlert, history } = this.props;

    return createPool(values).then(() => {
      showAlert({
        type: 'success',
        message: `Created IP pool ${values.name}.`
      });
      history.push('/account/ip-pools');
    });
  };

  renderError() {
    const { listError } = this.props;
    return <ApiErrorBanner
      errorDetails={listError.message}
      message="Sorry, we seem to have had some trouble loading your IP pool."
      reload={this.loadDependentData}
    />;
  }

  render() {
    const { loading, listError } = this.props;

    if (loading) {
      return <Loading />;
    }

    if (listError) {
      return this.renderError();
    }

    return (
      <Page title="Create IP Pool" breadcrumbAction={breadcrumbAction}>
        <PoolForm onSubmit={this.createPool} isNew={true} />
      </Page>
    );
  }
}

const mapStateToProps = (state, props) => ({
  listError: state.ipPools.listError,
  loading: state.ipPools.listLoading
});

export default connect(mapStateToProps, {
  createPool,
  listPools,
  showAlert
})(CreatePage);
