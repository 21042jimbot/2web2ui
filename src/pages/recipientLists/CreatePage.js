import React, { Component } from 'react';
import { connect } from 'react-redux';

import { Link, withRouter } from 'react-router-dom';

import { Page } from 'src/components/matchbox';

import { createRecipientList } from 'src/actions/recipientLists';
import { showAlert } from 'src/actions/globalAlert';

import RecipientListForm from './components/RecipientListForm';

export class CreatePage extends Component {
  createRecipientList = values => {
    const { createRecipientList, showAlert, history } = this.props;

    return createRecipientList(values).then(() => {
      showAlert({
        type: 'success',
        message: 'Created recipient list',
      });
      history.push('/lists/recipient-lists');
    });
  };

  render() {
    return (
      <Page
        title="Create Recipient List"
        breadcrumbAction={{
          content: 'Recipient Lists',
          Component: Link,
          to: '/lists/recipient-lists',
        }}
      >
        <RecipientListForm onSubmit={this.createRecipientList} />
      </Page>
    );
  }
}

const mapDispatchToProps = {
  createRecipientList,
  showAlert,
};

export default withRouter(connect(undefined, mapDispatchToProps)(CreatePage));
