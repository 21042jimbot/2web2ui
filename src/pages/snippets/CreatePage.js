import React from 'react';
import { connect } from 'react-redux';
import { Field, Form, reduxForm } from 'redux-form';
import { Button, Grid, Page, Panel } from '@sparkpost/matchbox';
import { createSnippet, getSnippet } from 'src/actions/snippets';
import { getSubaccount } from 'src/actions/subaccounts';
import ContentEditor from 'src/components/contentEditor';
import Loading from 'src/components/loading';
import PageLink from 'src/components/pageLink';
import TextFieldWrapper from 'src/components/reduxFormWrappers/TextFieldWrapper';
import SubaccountSection from 'src/components/subaccountSection';
import { slugify } from 'src/helpers/string';
import { maxLength, required, slug } from 'src/helpers/validation';
import { hasSubaccounts } from 'src/selectors/subaccounts';
import IdentifierHelpText from './components/IdentifierHelpText';

export class CreatePage extends React.Component {
  componentDidMount() {
    // Sent here via duplicate button
    if (this.props.location.state) {
      const { id, subaccount_id } = this.props.location.state;
      this.props.getSnippet({ id, subaccountId: subaccount_id });
    }
  }

  componentDidUpdate() {
    if (this.props.submitSucceeded) {
      this.props.history.push('/snippets');
    }
  }

  fillIdField = (event) => {
    this.props.change('id', slugify(event.target.value));
  }

  submitSnippet = ({
    assignTo,
    content: { html, text } = {},
    id,
    name,
    subaccount: { id: subaccountId } = {}
  }) => (
    this.props.createSnippet({
      html,
      id,
      name,
      sharedWithSubaccounts: assignTo === 'shared',
      subaccountId,
      text
    })
  )

  render() {
    const { handleSubmit, hasSubaccounts, submitting, loading } = this.props;

    if (loading) {
      return <Loading />;
    }

    return (
      <Page
        title="New Snippet"
        breadcrumbAction={{ Component: PageLink, content: 'Snippets', to: '/snippets' }}
        primaryAction={{
          Component: Button,
          content: 'Create Snippet',
          onClick: handleSubmit(this.submitSnippet)
        }}
      >
        <Form onSubmit={this.submitSnippet}>
          <Grid>
            <Grid.Column xs={12} lg={4}>
              <Panel>
                <Panel.Section>
                  <Field
                    name="name"
                    component={TextFieldWrapper}
                    disabled={submitting}
                    label="Snippet Name"
                    onChange={this.fillIdField}
                    validate={[required, maxLength(64)]}
                  />
                  <Field
                    name="id"
                    component={TextFieldWrapper}
                    disabled={submitting}
                    helpText={<IdentifierHelpText />}
                    label="Snippet ID"
                    validate={[required, slug, maxLength(64)]}
                  />
                </Panel.Section>
                {hasSubaccounts && <SubaccountSection newTemplate={true} disabled={submitting} />}
              </Panel>
            </Grid.Column>
            <Grid.Column xs={12} lg={8}>
              <ContentEditor contentOnly={true} />
            </Grid.Column>
          </Grid>
        </Form>
      </Page>
    );
  }
}

const formOptions = {
  form: 'createSnippetForm',
  enableReinitialize: true
};

const mapDispatchToProps = {
  createSnippet,
  getSnippet,
  getSubaccount
};

const mapStateToProps = (state, props) => ({
  hasSubaccounts: hasSubaccounts(state),
  initialValues: getInitialValues(state, props),
  loading: state.snippets.loading
});

const getInitialValues = (state, props) => {
  if (state.snippets.snippet) {
    const { id, name, content, shared_with_subaccounts } = state.snippets.snippet;
    const { subaccount_id } = props.location.state;
    return {
      id: `${id}-copy`,
      name: `${name} Copy`,
      subaccount: { id: subaccount_id },
      assignTo: subaccount_id ? 'subaccount' : shared_with_subaccounts ? 'shared' : 'master',
      content
    };
  } else {
    return {
      assignTo: 'master'
    };
  }
};

export default connect(mapStateToProps, mapDispatchToProps)(reduxForm(formOptions)(CreatePage));
