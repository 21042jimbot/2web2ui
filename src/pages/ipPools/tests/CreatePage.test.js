import { shallow } from 'enzyme';
import React from 'react';
import { CreatePage } from '../CreatePage';

describe('IP Pools Create Page', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      createPool: jest.fn(() => Promise.resolve()),
      showAlert: jest.fn(),
      history: {
        push: jest.fn()
      },
      loading: false
    };

    wrapper = shallow(<CreatePage {...props} />);
  });

  it('should render the list page correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should show loading component when data is loading', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('should show an alert on successful pool creation', async () => {
    await wrapper.instance().createPool({ name: 'my-pool' });
    expect(wrapper.instance().props.showAlert).toHaveBeenCalledWith({
      type: 'success',
      message: 'Created IP pool my-pool.'
    });
    expect(wrapper.instance().props.history.push).toHaveBeenCalled();

  });
});
