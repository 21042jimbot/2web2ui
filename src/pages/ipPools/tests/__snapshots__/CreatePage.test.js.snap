// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IP Pools Create Page should render the list page correctly 1`] = `
<Page
  breadcrumbAction={
    Object {
      "Component": [Function],
      "content": "IP Pools",
      "to": "/account/ip-pools",
    }
  }
  empty={Object {}}
  title="Create IP Pool"
>
  <Panel>
    <Panel.Section>
      <Connect(ReduxForm)
        isNew={true}
        onSubmit={[Function]}
      />
    </Panel.Section>
  </Panel>
</Page>
`;

exports[`IP Pools Create Page should show loading component when data is loading 1`] = `<Loading />`;
